src/app.js

```diff
@@ -1,35 +1,101 @@
-var JiraClient = require("jira-connector");
-
-function connectJira(action, settings) {
-  // Initialize
-  var jira = new JiraClient({
-    host: action.params.HOST || settings.HOST,
-    basic_auth: {
-      email: action.params.EMAIL || settings.EMAIL,
-      api_token: action.params.TOKEN || settings.TOKEN
-    },
-    strictSSL: true // One of optional parameters
-  });
-  return jira
+const fetch = require('node-fetch');
+
+function updateIssueStatus(action, settings) {
+    /**
+     * Change the status of an issue.
+     */
+    let jHost = action.params.HOST || settings.HOST;
+    const jIssueKey = action.params.ISSUE_KEY;
+    const jStatusID = action.params.STATUS_ID;
+    const method = "POST"
+    jHost = `${jHost}/rest/api/3/issue/${jIssueKey}/transitions?expand=transitions.fields`;
+    let bodyData = `{
+      "transition": {
+        "id":"${jStatusID}"            
+      }
+    }`
+    return genericRestAPI(action,settings,method, jHost, bodyData)
 }
 
+async function updateVersion (action, settings) {
+  /**
+   * Set status of version to released inside a given project.
+   */
+  const host= action.params.HOST || settings.HOST;
+  const projectId = action.params.PROJECT_ID;
+  const versionId = await getVersionId(action, settings, action.params.VERSION, projectId,host);
+  const jHost = `${host}/rest/api/3/version/${versionId}`
+  const intProjectId = parseInt(projectId);
+  const method = "PUT"
+  let bodyData = `{
```

Comment:
@ilanyaniv change `bodyData` all across to be objects and use `JSON.stringify` to convert to string in `genericRestAPI` method.

Reply:
@kadosh1000 
I replaced the strings into objects. Also replaced the zip file and cleaned DS files.